gc/gc-base/src/test/java/org/projectnessie/gc/base/TestInstantConversion.java

```diff
@@ -17,23 +17,62 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 
+import java.time.Clock;
 import java.time.Instant;
+import java.time.ZoneId;
+import java.time.temporal.ChronoUnit;
+import java.util.TimeZone;
 import java.util.concurrent.TimeUnit;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 
 public class TestInstantConversion {
 
-  @Test
-  void testInstantConversion() {
-    Instant now = Instant.now();
+  @ParameterizedTest
+  @ValueSource(booleans = {false, true})
+  void testInstantConversion(boolean isNanoPrecision) {
+    Instant now = isNanoPrecision ? Instant.now(new NanoClock(Clock.systemUTC())) : Instant.now();
     long time = now.getEpochSecond();
     long nano = now.getNano();
     // current time in microseconds since epoch
     long micro = TimeUnit.SECONDS.toMicros(time) + TimeUnit.NANOSECONDS.toMicros(nano);
     Instant instantFromMicros = GCUtil.getInstantFromMicros(micro);
-    // Even though instant is capable of nanosecond precision,
-    // Instant.now() gives micro second precision by default.
-    // Because of that the below validation can pass.
-    assertThat(instantFromMicros).isEqualTo(now);
+    // Instant is capable of nanosecond precision.
+    // Hence, convert to microsecond precision before comparing.
+    Instant instantInMicros = now.truncatedTo(ChronoUnit.MICROS);
+    assertThat(instantFromMicros).isEqualTo(instantInMicros);
+  }
+
+  private static class NanoClock extends Clock {
```

Comment:
I added this for java 11 handling the nano second instants.

Reply:
I don't understand why this is needed for the test.